stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -50,22 +85,70 @@ internal class PaymentSheetViewModel internal constructor(
     }
 
     fun updatePaymentMethods(intent: Intent) {
-        val args: PaymentSheetActivityStarter.Args? = PaymentSheetActivityStarter.Args.fromIntent(intent)
-        if (args == null) {
-            onError(IllegalStateException("Missing activity args"))
-        } else {
+        getPaymentSheetActivityArgs(intent)?.let { args ->
             updatePaymentMethods(
                 args.ephemeralKey,
                 args.customerId
             )
         }
     }
 
+    fun checkout(activity: Activity) {
+        val args = getPaymentSheetActivityArgs(activity.intent) ?: return
```

Comment:
Should this call `onError` instead of simply returning?

Reply:
`getPaymentSheetActivityArgs` will call onError. I will admit that's a bit of a confusing pattern though. Maybe it's simpler to just duplicate the logic for calling `onError`?